### PR TITLE
Fixes the man pages creation

### PR DESCRIPTION
**Pull request by Gu1nness**
The issue https://github.com/borgbackup/borg/issues/4471 solution
created many commands for borgfs which shouldn't exist in addition to
creating issue https://github.com/borgbackup/borg/issues/4752.

Fixes https://github.com/borgbackup/borg/issues/4752 .

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -2604,6 +2604,22 @@ def define_archive_filters_group(subparser, *, sort_by=True, first_last=True):
                 group.add_argument('--last', metavar='N', dest='last', default=0, type=positive_int_validator,
                                    help='consider last N archives after other filters were applied')
 
+        def define_borg_mount(subparser):
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: guess this should be `parser`.

that the parser can be in fact a subparser sometimes is a detail not needed here (and also is not true for borgfs).

File: src/borg/archiver.py [hunk 2]
```diff
@@ -2623,6 +2639,55 @@ def define_archive_filters_group(subparser, *, sort_by=True, first_last=True):
...
+        signals or crashes do not.
+        """)
+
+        if parser.prog == 'borgfs':
+            parser.description = self.do_mount.__doc__
+            parser.epilog = mount_epilog
+            parser.formatter_class = argparse.RawDescriptionHelpFormatter
+            parser.help = 'mount repository'
+            subparser = parser
+            define_borg_mount(subparser)
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: replace these 2 lines by:
```
define_borg_mount(parser)
```

The introduction of the `subparser` name is a relict from the old code structure - but as you have it as a function paramenter now, it is not needed any more.

**Comment by fantasya-pbem** [comment]
Oh dear, guess that is my fault, the fix for issue #4471 was mine...
I can confirm that build_man now builds successful.

**Comment by Gu1nness** [comment]
Indeed most of the patch is moving the code around.
I patched your remarks, I was tired when I wrote this it seems :)

**Comment by ThomasWaldmann** [comment]
Thanks!

Is this change only needed for master or also for 1.1-maint?

**Comment by ThomasWaldmann** [comment]
Also, see #4789, there might be another issue.

**Comment by Gu1nness** [comment]
It does not seem to be necessary for 1.1-maint the man pages build correctly.
For #4789 the man page does not seem to list the commands especially (except in the "SEE ALSO" section, in which they are alphabetically unordered (but logically ordered : 
```man
borg-init(1), borg-create(1), borg-mount(1),  borg-extract(1),  borg-list(1),  borg-info(1),  borg-delete(1),  borg-prune(1), borg-recreate(1
```

**Comment by ThomasWaldmann** [comment]
1.1-maint: ok, good.

what i meant with #4789 issue is that the borg.1 manpage is not updated.

**Comment by Gu1nness** [comment]
Oh yeah indeed.
Having a look at the code, I can see [here](https://github.com/borgbackup/borg/blob/master/docs/man_intro.rst) that the date is hardcoded in the man_intro.
Let me patch this on my fork and I'll give you the corresponding commit :)

**Pull request merged**